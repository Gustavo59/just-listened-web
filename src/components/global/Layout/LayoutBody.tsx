import { ReactNode } from "react";

type LayoutBodyProps = {
    children: ReactNode;
};

export function LayoutBody(props: LayoutBodyProps) {
    return <div>{props.children}</div>;
}
