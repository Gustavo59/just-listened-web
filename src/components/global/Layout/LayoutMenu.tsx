import Icon, {
    HomeOutlined,
    PoweroffOutlined,
    SearchOutlined,
} from "@ant-design/icons";
import type { MenuProps } from "antd";
import { Button, Menu, Image } from "antd";
import { useRouter } from "next/router";
import { signOut, useSession } from "next-auth/react";

import { AlbumsIcon } from "@/components/icons/AlbumsIcon";
import { ArtistsIcon } from "@/components/icons/ArtistsIcon";
import { CurrentSession } from "@/domain/entities/CurrentSession";

export function LayoutMenu() {
    const router = useRouter();

    const handleMenuClick = (destPath: string) => {
        router.push(destPath);
    };

    const { data: session } = useSession() as CurrentSession;

    const items: MenuProps["items"] = [
        {
            label: "Home",
            key: "/",
            icon: <HomeOutlined />,
        },
        {
            label: "Albums",
            key: "/albums",
            icon: <AlbumsIcon />,
            disabled: true,
        },
        {
            label: "Artists",
            key: "/artists",
            icon: <ArtistsIcon />,
            disabled: true,
        },
        {
            label: "Search",
            key: "/search",
            icon: <SearchOutlined />,
        },
        {
            label: `${session.user.name}`,
            key: "/user",
            icon: (
                <Icon
                    component={() => (
                        <Image
                            preview={false}
                            height={30}
                            src={session.user.image}
                            style={{ borderRadius: "15px" }}
                            alt="user-image"
                        />
                    )}
                />
            ),
            style: {
                marginLeft: "auto",
            },
            disabled: true,
        },
        {
            label: (
                <Button
                    icon={<PoweroffOutlined />}
                    type="text"
                    onClick={() => signOut()}
                    danger
                >
                    Sign Out
                </Button>
            ),
            key: "options",
        },
    ];

    return (
        <>
            <Menu
                mode="horizontal"
                items={items}
                onClick={(e) => {
                    handleMenuClick(e.key);
                }}
            />
        </>
    );
}
