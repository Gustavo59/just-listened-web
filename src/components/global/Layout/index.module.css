div.Layout > section {
    background: white;
}

section.Layout__Menu > img {
    border-radius: 30px;
}

section.Layout__Body {
    grid-area: body;
}

section.Layout__Body > div {
    margin: 20px auto auto 20px;
    width: 95%;
}
