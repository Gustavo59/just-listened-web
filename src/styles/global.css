@import "swiper/css";
@import "swiper/css/navigation";
@import "swiper/css/pagination";

* {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
}

html,
body {
    min-height: 100vh;
    max-width: 100vw;
    overflow-x: hidden;
}

#__next {
    min-height: 100vh;
}

.ant-space {
    width: 100%;
}

.ant-card:hover {
    box-shadow: 0 0 10px #7dd96a;
    transition:
        box-shadow 0.2s ease-out,
        border-color 0.2s ease-out;
}

.swiper-button-prev,
.swiper-button-next {
    color: white !important;
    background-color: #7dd96a !important;
    height: 25px !important;
    width: 25px !important;
    border-radius: 15px !important;
}

.swiper-button-prev:after,
.swiper-button-next:after {
    font-size: 15px !important;
    font-weight: bold !important;
}
