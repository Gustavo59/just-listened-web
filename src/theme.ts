import { ThemeConfig } from "antd";

const config: ThemeConfig = {
    token: {
        colorPrimary: "#7dd96a",
    },
};

export default config;
