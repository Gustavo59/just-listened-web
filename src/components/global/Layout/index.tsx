import { ReactNode } from "react";

import styles from "./index.module.css";
import { useRouter } from "next/router";
import { LayoutMenu } from "./LayoutMenu";
import { LayoutBody } from "./LayoutBody";

type LayoutProps = {
    children: ReactNode;
};

export function Layout(props: LayoutProps) {
    const router = useRouter();

    if (router.pathname === "/login") {
        return <>{props.children}</>;
    }
    return (
        <div className={styles.Layout}>
            <section className={styles.Layout__Menu}>
                <LayoutMenu />
            </section>
            <section className={styles.Layout__Body}>
                <LayoutBody>{props.children}</LayoutBody>
            </section>
        </div>
    );
}
