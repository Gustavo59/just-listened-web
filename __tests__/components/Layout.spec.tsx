import { Layout } from "@/components/global/Layout";
import { render, waitFor } from "@testing-library/react";
import { ConfigProvider } from "antd";

import theme from "@/theme";
import { useSession } from "next-auth/react";
import { NEXT_AUTH_SESSION_DATA } from "../data/libs/next-auth";
import { useRouter } from "next/router";

jest.mock("next-auth/react", () => ({ useSession: jest.fn() }));
jest.mock("next/router", () => ({ useRouter: jest.fn() }));

describe("layout component", () => {
    test("renders the layout", async () => {
        (useRouter as jest.Mock).mockReturnValue({});
        (useSession as jest.Mock).mockReturnValue(NEXT_AUTH_SESSION_DATA);

        const { container } = render(
            <ConfigProvider theme={theme}>
                <Layout>
                    <h1>Test Layout</h1>
                </Layout>
            </ConfigProvider>,
        );

        await waitFor(() => {
            expect(
                container.getElementsByClassName("ant-menu-item").length,
            ).toBe(6);
            expect(container).toMatchSnapshot();
        });
    });
});
