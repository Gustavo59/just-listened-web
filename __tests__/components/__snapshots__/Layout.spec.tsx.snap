// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout component renders the layout 1`] = `
<div>
  <div
    class="Layout"
  >
    <section
      class="Layout__Menu"
    >
      <ul
        class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light css-dev-only-do-not-override-13pm87f"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-selected"
          data-menu-id="rc-menu-uuid-test-/"
          role="menuitem"
          style="opacity: 1; order: 0;"
          tabindex="-1"
        >
          <span
            aria-label="home"
            class="anticon anticon-home ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Home
          </span>
        </li>
        <li
          aria-disabled="true"
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-disabled"
          role="menuitem"
          style="opacity: 1; order: 1;"
        >
          <span
            class="anticon"
            role="img"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M425.706,86.294A240,240,0,0,0,86.294,425.706,240,240,0,0,0,425.706,86.294ZM256,464C141.309,464,48,370.691,48,256S141.309,48,256,48s208,93.309,208,208S370.691,464,256,464Z"
                fill="currentColor"
              />
              <path
                d="M256,152A104,104,0,1,0,360,256,104.118,104.118,0,0,0,256,152Zm0,176a72,72,0,1,1,72-72A72.081,72.081,0,0,1,256,328Z"
                fill="currentColor"
              />
              <rect
                fill="currentColor"
                height="32"
                width="32"
                x="240"
                y="240"
              />
              <path
                d="M256,112V80a174.144,174.144,0,0,0-79.968,19.178A177.573,177.573,0,0,0,115.2,150.39l25.586,19.219A142.923,142.923,0,0,1,256,112Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Albums
          </span>
        </li>
        <li
          aria-disabled="true"
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-disabled"
          role="menuitem"
          style="opacity: 1; order: 2;"
        >
          <span
            class="anticon"
            role="img"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 12.25C13.2583 12.25 12.5333 12.0301 11.9166 11.618C11.2999 11.206 10.8193 10.6203 10.5355 9.93506C10.2516 9.24984 10.1774 8.49584 10.3221 7.76841C10.4668 7.04098 10.8239 6.3728 11.3484 5.84835C11.8728 5.3239 12.541 4.96675 13.2684 4.82206C13.9958 4.67736 14.7498 4.75162 15.4351 5.03545C16.1203 5.31928 16.706 5.79993 17.118 6.41661C17.5301 7.0333 17.75 7.75832 17.75 8.5C17.75 9.49456 17.3549 10.4484 16.6517 11.1517C15.9484 11.8549 14.9946 12.25 14 12.25ZM14 6.25C13.555 6.25 13.12 6.38196 12.75 6.62919C12.38 6.87643 12.0916 7.22783 11.9213 7.63896C11.751 8.0501 11.7064 8.5025 11.7932 8.93895C11.8801 9.37541 12.0943 9.77632 12.409 10.091C12.7237 10.4057 13.1246 10.62 13.561 10.7068C13.9975 10.7936 14.4499 10.749 14.861 10.5787C15.2722 10.4084 15.6236 10.12 15.8708 9.75003C16.118 9.38002 16.25 8.94501 16.25 8.5C16.25 7.90326 16.0129 7.33097 15.591 6.90901C15.169 6.48705 14.5967 6.25 14 6.25Z"
                fill="currentColor"
              />
              <path
                d="M21 19.25C20.8019 19.2474 20.6126 19.1676 20.4725 19.0275C20.3324 18.8874 20.2526 18.6981 20.25 18.5C20.25 16.55 19.19 15.25 14 15.25C8.81 15.25 7.75 16.55 7.75 18.5C7.75 18.6989 7.67098 18.8897 7.53033 19.0303C7.38968 19.171 7.19891 19.25 7 19.25C6.80109 19.25 6.61032 19.171 6.46967 19.0303C6.32902 18.8897 6.25 18.6989 6.25 18.5C6.25 13.75 11.68 13.75 14 13.75C16.32 13.75 21.75 13.75 21.75 18.5C21.7474 18.6981 21.6676 18.8874 21.5275 19.0275C21.3874 19.1676 21.1981 19.2474 21 19.25Z"
                fill="currentColor"
              />
              <path
                d="M8.31999 13.06H7.99999C7.20434 12.9831 6.47184 12.5933 5.96361 11.9763C5.45539 11.3593 5.21308 10.5657 5.28999 9.77001C5.36691 8.97436 5.75674 8.24186 6.37373 7.73363C6.99073 7.22541 7.78434 6.9831 8.57999 7.06001C8.68201 7.0644 8.78206 7.08957 8.87401 7.13399C8.96596 7.1784 9.04787 7.24113 9.11472 7.31831C9.18157 7.3955 9.23196 7.48553 9.26279 7.58288C9.29362 7.68023 9.30425 7.78285 9.29402 7.88445C9.28379 7.98605 9.25292 8.08449 9.20331 8.17374C9.15369 8.26299 9.08637 8.34116 9.00548 8.40348C8.92458 8.46579 8.83181 8.51093 8.73286 8.53613C8.6339 8.56133 8.53084 8.56605 8.42999 8.55001C8.23479 8.53055 8.03766 8.55062 7.85038 8.60904C7.6631 8.66746 7.48952 8.76302 7.33999 8.89001C7.18812 9.01252 7.06216 9.16403 6.96945 9.33572C6.87673 9.50741 6.81913 9.69583 6.79999 9.89001C6.77932 10.0866 6.79797 10.2854 6.85488 10.4747C6.91178 10.6641 7.0058 10.8402 7.13144 10.9928C7.25709 11.1455 7.41186 11.2716 7.58673 11.3638C7.76159 11.456 7.95307 11.5125 8.14999 11.53C8.47553 11.5579 8.80144 11.4808 9.07999 11.31C9.24973 11.2053 9.45413 11.1722 9.64824 11.2182C9.84234 11.2641 10.0102 11.3853 10.115 11.555C10.2198 11.7248 10.2528 11.9292 10.2069 12.1233C10.1609 12.3174 10.0397 12.4853 9.86999 12.59C9.40619 12.8858 8.86998 13.0484 8.31999 13.06Z"
                fill="currentColor"
              />
              <path
                d="M3 18.5C2.80189 18.4974 2.61263 18.4176 2.47253 18.2775C2.33244 18.1374 2.25259 17.9481 2.25 17.75C2.25 15.05 2.97 13.25 6.5 13.25C6.69891 13.25 6.88968 13.329 7.03033 13.4697C7.17098 13.6103 7.25 13.8011 7.25 14C7.25 14.1989 7.17098 14.3897 7.03033 14.5303C6.88968 14.671 6.69891 14.75 6.5 14.75C4.15 14.75 3.75 15.5 3.75 17.75C3.74741 17.9481 3.66756 18.1374 3.52747 18.2775C3.38737 18.4176 3.19811 18.4974 3 18.5Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Artists
          </span>
        </li>
        <li
          aria-disabled="true"
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-disabled"
          role="menuitem"
          style="opacity: 1; order: 3;"
        >
          <span
            aria-label="search"
            class="anticon anticon-search ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Search
          </span>
        </li>
        <li
          aria-disabled="true"
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-disabled"
          role="menuitem"
          style="opacity: 1; order: 4; margin-left: auto;"
        >
          <span
            class="anticon ant-menu-item-icon"
            role="img"
          >
            <div
              class="ant-image css-dev-only-do-not-override-13pm87f"
              style="height: 30px;"
            >
              <img
                alt="user-image"
                class="ant-image-img css-dev-only-do-not-override-13pm87f"
                height="30"
                src="https://example.com/"
                style="height: 30px; border-radius: 15px;"
              />
            </div>
          </span>
          <span
            class="ant-menu-title-content"
          >
            John Doe
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
          role="menuitem"
          style="opacity: 1; order: 5;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-13pm87f ant-btn-text ant-btn-dangerous"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <span
                  aria-label="poweroff"
                  class="anticon anticon-poweroff"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="poweroff"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M705.6 124.9a8 8 0 00-11.6 7.2v64.2c0 5.5 2.9 10.6 7.5 13.6a352.2 352.2 0 0162.2 49.8c32.7 32.8 58.4 70.9 76.3 113.3a355 355 0 0127.9 138.7c0 48.1-9.4 94.8-27.9 138.7a355.92 355.92 0 01-76.3 113.3 353.06 353.06 0 01-113.2 76.4c-43.8 18.6-90.5 28-138.5 28s-94.7-9.4-138.5-28a353.06 353.06 0 01-113.2-76.4A355.92 355.92 0 01184 650.4a355 355 0 01-27.9-138.7c0-48.1 9.4-94.8 27.9-138.7 17.9-42.4 43.6-80.5 76.3-113.3 19-19 39.8-35.6 62.2-49.8 4.7-2.9 7.5-8.1 7.5-13.6V132c0-6-6.3-9.8-11.6-7.2C178.5 195.2 82 339.3 80 506.3 77.2 745.1 272.5 943.5 511.2 944c239 .5 432.8-193.3 432.8-432.4 0-169.2-97-315.7-238.4-386.7zM480 560h64c4.4 0 8-3.6 8-8V88c0-4.4-3.6-8-8-8h-64c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                Sign Out
              </span>
            </button>
          </span>
        </li>
        <li
          aria-hidden="true"
          class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
          role="none"
          style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
        >
          <div
            aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            data-menu-id="rc-menu-uuid-test-rc-menu-more"
            role="menuitem"
            tabindex="-1"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      />
    </section>
    <section
      class="Layout__Body"
    >
      <div>
        <h1>
          Test Layout
        </h1>
      </div>
    </section>
  </div>
</div>
`;
